### DISCUSSION TITLE
Does this library have `tool_choice` option?

### DISCUSSION THREAD

**Discussion by avergin** [brainlid/langchain#140] [Q&A]
Does this library have the `tool_choice` option when sending requests to models that have this functionality (e.g. openai and anthropic)? If this is something currently not supported, are there plans in the future to support this?

`tool_choice` options are as follows for the model providers:
- OpenAI: `none`, `auto` (default), `required`, `{"type": "function", "function": {"name": "my_function"}}` [docs](https://platform.openai.com/docs/guides/function-calling/function-calling-behavior)
- Anthropic: `none`, `any`, `{"type": "tool", "name": "get_weather"}` [docs](https://docs.anthropic.com/en/docs/build-with-claude/tool-use#forcing-tool-use)

**Comment by brainlid** [comment]
Hi @avergin! Forcing a `tool_choice` is not currently supported but should be fairly trivial to add. It's not currently on my short-list of things to do but I'd welcome a contribution. I don't have any immediate plans to add it but I'm not opposed to it either. :slightly_smiling_face:

**Reply by avergin** [reply]
Opened a PR draft that adds `tool_choice` to both OpenAI and Anthropic https://github.com/brainlid/langchain/pull/142
